### ISSUE TITLE
Enhancement: add temp reading of fritz box

### ISSUE THREAD

**Issue by RenGro** [andig/pimatic-fritz#2]
Hi Andig,

first of all: it's a really nice plugin to control Fritz!Dect by Pimatic.
State, power and energy attributes are working fine.
Since some weeks Fritz!Dect is able to measure and display the temperature within offical firmware.

Would you be able to enhance your plugin, so that the temperature could also be used within Pimatic?

Thanks in advance
RenGro

**Comment by andig** [comment]
Will do asap. Do you by chance have a pointer to the current api docs?

**Comment by RenGro** [comment]
Hi Andig,

Sounds great!

But I cannot understand your question :-(

Maybe you can write in German :-)

Von meinem iPad gesendet

**Comment by andig** [comment]
Ich finde bei Avm keine Doku wie die Temperatur auszulesen ist?

**Comment by RenGro** [comment]
Hi,
 
habe Google mal gequält und auch nicht wirklich viel Infos gefunden.
 
Über http://www.tdressler.net/ipsymcon/fritz_aha.html bin ich auf die AVM Schnittstellendokus gestoßen.
Und dort steht, dass über getdevicelistinfos die Temperatur ausgelesen werden kann (mittels AHA HTTP Interface)
 
Grüße
René
 
-----Original-Nachricht-----
Betreff: Re: [pimatic-fritz] Enhancement: (#2)
Datum: Sun, 28 Dec 2014 19:59:49 +0100
Von: andig notifications@github.com
An: andig/pimatic-fritz pimatic-fritz@noreply.github.com
 
 

Ich finde bei Avm keine Doku wie die Temperatur auszulesen ist?

**Comment by mwittig** [comment]
According to http://avm.de/fileadmin/user_upload/Global/Service/Schnittstellen/AHA-HTTP-Interface.pdf the temperature can only be read using the "getdevicelistinfos" method. As this method also provides switch state, power, energy values, an update can now be obtained with a single fritzcall rather than performing three calls per update cycle. However, as "getdevicelistinfos" returns XML I had to integrate an XML Parser. I have chosen xml2js to map the XML document to a simplified JSON object. This way, the code is easier to read a no xpath expressions are required for scoping. 

I have also added support for xAttributeOptions and xLink extensions
